synapse/handlers/room_member.py

```diff
@@ -830,10 +830,17 @@ async def update_membership_locked(
 
                 content["membership"] = Membership.JOIN
 
-                profile = self.profile_handler
-                if not content_specified:
-                    content["displayname"] = await profile.get_displayname(target)
-                    content["avatar_url"] = await profile.get_avatar_url(target)
+                try:
+                    profile = self.profile_handler
+                    if not content_specified:
+                        content["displayname"] = await profile.get_displayname(target)
+                        content["avatar_url"] = await profile.get_avatar_url(target)
+                except Exception as e:
+                    logger.info(
+                        "Failed to get profile information while processing remote join for %r: %s",
+                        target,
+                        e,
+                    )
```

Comment:
Same pattern that we already use in https://github.com/matrix-org/synapse/blob/e714b8a057f65fe07b4f3939e018e57862980cdc/synapse/handlers/message.py#L640-L652